extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -863,79 +977,173 @@ public static JsonArray getJsonResponse(@Nonnull final Page page,
+                        .value("platform", "MOBILE")
+                        .value("hl", localization.getLocalizationCode())
+                        .value("gl", contentCountry.getCountryCode())
                     .end()
-                    .done())
-                    .getBytes(UTF_8);
-            // @formatter:on
-        } else {
-            // @formatter:off
-            return JsonWriter.string(prepareDesktopJsonBuilder(localization, contentCountry)
-                    .value("videoId", videoId)
-                    .object("playbackContext")
-                        .object("contentPlaybackContext")
-                            .value("signatureTimestamp", sts)
+                    .object("thirdParty")
+                        .value("embedUrl", "https://www.youtube.com/watch?v=" + videoId)
+                    .end()
+                    .object("request")
+                        .array("internalExperimentFlags")
                         .end()
+                        .value("useSsl", true)
                     .end()
-                    .done())
-                    .getBytes(UTF_8);
-            // @formatter:on
-        }
+                    .object("user")
+                        // TO DO: provide a way to enable restricted mode with:
+                        // .value("enableSafetyMode", boolean)
+                        .value("lockedSafetyMode", false)
+                    .end()
+                .end()
+                .value(CPN, contentPlaybackNonce)
+                .value(VIDEO_ID, videoId);
+        // @formatter:on
+    }
+
+    @Nonnull
+    public static byte[] createDesktopPlayerBody(
+            @Nonnull final Localization localization,
+            @Nonnull final ContentCountry contentCountry,
+            @Nonnull final String videoId,
+            @Nonnull final String sts,
+            final boolean isEmbedClientScreen,
+            @Nonnull final String contentPlaybackNonce) throws IOException, ExtractionException {
+        // @formatter:off
+        return JsonWriter.string((isEmbedClientScreen
+                        ? prepareDesktopEmbedVideoJsonBuilder(localization, contentCountry,
+                            videoId)
+                        : prepareDesktopJsonBuilder(localization, contentCountry))
+                .object("playbackContext")
+                    .object("contentPlaybackContext")
+                        .value("currentUrl", "/watch?v=" + videoId)
+                        .value("vis", 0)
+                        .value("splay", false)
+                        .value("autoCaptionsDefaultOn", false)
+                        .value("autonavState", "STATE_NONE")
+                        .value("html5Preference", "HTML5_PREF_WANTS")
+                        .value("signatureTimestamp", sts)
+                        .value("referer", "https://www.youtube.com/watch?v=" + videoId)
+                        .value("lactMilliseconds", "-1")
```

Comment:
It would be great to know what these parameters do (e.g. ``lactMilliseconds``) or where they are coming from.

Reply:
These parameters comes from a player request by an official client (the `WEB` one, of course).

Probably some of them are unneeded to be sure to avoid throttling on streaming URLs of the `WEB` client, but I don't know them, so this should be investigated.

Reply:
> These parameters comes from a player request by an official client (the WEB one, of course).

Ok, please also comment this inside the code.

> Probably some of them are unneeded to be sure to avoid throttling on streaming URLs of the WEB client, but I don't know them, so this should be investigated.

👍 (consider creating a new issue if you don't manage to investigate them here 😄)

Reply:
I removed most of the parameters, except `signatureTimestamp` (essential) and `referer`. We will have to see if the buffering issue still occurs after this PR is merged, and then if necessary, readd some of them.